LFS171x/docs/introduction-to-hyperledger-fabric.md

```diff
@@ -1 +1,1547 @@
+
+err := stub.PutState(args[0], []byte(args[1]))
+
+if err != nil {
+
+return shim.Error(fmt.Sprintf("Failed to create asset: %s", args[0]))
+
+}
+
+return shim.Success(nil)
+
+}
+```
+
+The Init implementation accepts two parameters as inputs, and proposes to write a key/value pair to the ledger by using the **stub.PutState** function. **GetStringArgs** retrieves and checks the validity of arguments which we expect to be a key/value pair. Therefore, we check to ensure that there are two arguments specified. If not, we return an error from the **Init** method, to indicate that something went wrong. Once we have verified the correct number of arguments, we can store the initial state in the ledger. In order to accomplish this, we call the **stub.PutState** function, specifying the first argument as the key, and the second argument as the value for that key. If no errors are returned, we will return success from the **Init** method.
+
+### Sample Chaincode Decomposed - Invoke Method
+
+Now, we’ll explore the **Invoke** method, which gets called when a transaction is proposed by a client application. In our sample, we will either get the value for a given asset, or propose to update the value for a specific asset.
+
+```golang
+func (t *SampleChaincode) Invoke(stub shim.ChaincodeStubInterface) peer.Response {
+
+// Extract the function and args from the transaction proposal
+
+fn, args := stub.GetFunctionAndParameters()
+
+var result string
+
+var err error
+
+if fn == "set" {
+
+result, err = set(stub, args)
+
+} else { // assume 'get' even if fn is nil
+
+result, err = get(stub, args)
+
+}
+
+if err != nil { //Failed to get function and/or arguments from transaction proposal
+
+return shim.Error(err.Error())
+
+}
+
+// Return the result as success payload
+
+return shim.Success([]byte(result))
+
+}
+```
+
+There are two basic actions a client can invoke: _get_ and _set_.
+
+ - The _get_ method will be used to query and return the value of an existing asset.
+ - The _set_ method will be used to create a new asset or update the value of an existing asset.
+
+To start, we’ll call **GetFunctionandParameters** to isolate the function name and parameter variables. Each transaction is either a set or a get. Let's first look at how the set method is implemented:
+
+```golang
+func set(stub shim.ChaincodeStubInterface, args []string) (string, error) {
+
+if len(args) != 2 {
+
+return "", fmt.Errorf("Incorrect arguments. Expecting a key and a value")
+
+}
+
+err := stub.PutState(args[0], []byte(args[1]))
+
+if err != nil {
+
+return "", fmt.Errorf("Failed to set asset: %s", args[0])
+
+}
+
+return args[1], nil
+
+}
+``` 
+
+The set method will create or modify an asset identified by a key with the specified value. The set method will modify the world state to include the key/value pair specified. If the key exists, it will override the value with the new one, using the **PutState** method; otherwise, a new asset will be created with the specified value.
```

Comment:
_set_ should be italicized twice in this line.
**`PutState`** should be bold code.